lib/src/tool_runner.dart

```diff
@@ -19,60 +19,49 @@ typedef FakeResultCallback = String Function(String tool,
 /// limiting both parallelization and the number of open temporary files.
 final MultiFutureTracker _toolTracker = new MultiFutureTracker(4);
 
-/// A helper class for running external tools.
-class ToolRunner {
-  /// Creates a new ToolRunner.
-  ///
-  /// Takes a [toolConfiguration] that describes all of the available tools.
-  /// An optional `errorCallback` will be called for each error message
-  /// generated by the tool.
-  ToolRunner(this.toolConfiguration, [this._errorCallback]);
+/// Can be called when the ToolRunner is no longer needed.
+///
+/// This will remove any temporary files created by the tool runner.
+class ToolTempFileTracker {
+  final Directory temporaryDirectory;
 
-  final ToolConfiguration toolConfiguration;
-  final ToolErrorCallback _errorCallback;
-  int _temporaryFileCount = 0;
+  ToolTempFileTracker._() : temporaryDirectory = Directory.systemTemp.createTempSync('dartdoc_tools_');
 
-  Future<Directory> _temporaryDirectory;
-  Future<Directory> get temporaryDirectory {
-    if (_temporaryDirectory == null) {
-      _temporaryDirectory = Directory.systemTemp.createTemp('dartdoc_tools_');
-    }
-    return _temporaryDirectory;
-  }
+  static ToolTempFileTracker _instance;
+  static ToolTempFileTracker get instance => _instance ??= ToolTempFileTracker._();
 
-  void _error(String message) {
-    if (_errorCallback != null) {
-      _errorCallback(message);
-    }
-  }
-
-  Future<File> _createTemporaryFile() async {
+  int _temporaryFileCount = 0;
+  Future<File> createTemporaryFile() async {
     _temporaryFileCount++;
     File tempFile = new File(pathLib.join(
```

Comment:
Couldn't this just use `createTemp` on a Directory to avoid needing to keep track of the `_temporaryFileCount`? I suppose it doesn't matter that much, but it seemed like what `createTemp` exists to do.

Reply:
I suppose it could.  createTemp however is a bit overkill as it is intended to prevent collisions between applications, not just within the same application.  Having the integer numbers in the name I actually like, since it makes it a little easier to see what's going on under the hood.